In [1]:
import json
data = [json.loads(line) for line in open('./face_recognition_annotations.json', 'r')]

In [2]:
print(data[0]['_via_img_metadata']['img_001.jpeg43329']['regions'])

[{'shape_attributes': {'name': 'rect', 'x': 98, 'y': 22, 'width': 104, 'height': 116}, 'region_attributes': {'Classes': {'Right': True}}}, {'shape_attributes': {'name': 'rect', 'x': 247, 'y': 43, 'width': 108, 'height': 132}, 'region_attributes': {'Classes': {'Left': True}}}]


In [3]:
print(data[0]['_via_img_metadata']['img_001.jpeg43329']['regions'][0])

{'shape_attributes': {'name': 'rect', 'x': 98, 'y': 22, 'width': 104, 'height': 116}, 'region_attributes': {'Classes': {'Right': True}}}


In [4]:
widths_list = []
heights_list = []
for key, value in data[0]['_via_img_metadata'].items():
    local_w = [k['shape_attributes']['width'] for k in value['regions']]
    widths_list.extend(local_w)
    local_h = [k['shape_attributes']['height'] for k in value['regions']]
    heights_list.extend(local_h)

In [5]:
print(widths_list)

[104, 108, 82, 125, 89, 117, 75, 121, 111, 79, 55, 61, 53, 100, 89, 77, 51, 48, 61, 69, 68, 147, 201, 121, 117, 115, 118, 99, 130, 74, 85, 183, 93, 96, 186, 146, 198, 67, 61, 84, 114, 60, 89, 108, 79, 54, 62, 61, 45, 61, 63, 57, 65, 83, 121, 113, 171, 182, 80, 49, 71, 79, 81, 59, 61, 82, 59, 50, 116, 66, 82, 76, 91, 130, 97, 46, 44, 75, 42, 53, 78, 84, 72, 71, 83, 67, 69, 110, 75, 61, 126, 43, 38, 36, 33, 40, 71, 61, 70, 50, 64, 56, 74, 93, 88, 108, 92, 83, 66, 78, 74, 87, 94, 98, 89, 71, 120, 113, 103, 96, 32, 41, 41, 74, 64, 81, 64, 55, 52, 83, 106, 112, 47, 64, 62, 41, 48, 67, 80, 47, 54, 75, 49, 59, 53, 121, 95, 100, 127, 118, 111, 51, 67, 59, 43, 85, 81, 71, 65, 60, 75, 77, 47, 29, 30, 25, 31, 76, 81, 73, 143, 92, 86, 53, 50, 57, 72, 113, 110, 85, 72, 41, 54, 38, 42, 53, 25, 75, 61, 38, 54, 85, 83, 50, 75, 72, 53, 69, 74, 73, 73, 78, 95, 120, 30, 57, 63, 60, 49, 86, 89, 102, 59, 90, 93, 97, 110, 108, 46, 55, 56, 51, 56, 64, 53, 51, 53, 49, 58, 90, 44, 42, 73, 78, 73, 87, 66, 82, 5

In [6]:
print(heights_list)

[116, 132, 106, 151, 118, 134, 64, 89, 127, 112, 73, 110, 124, 79, 86, 86, 101, 94, 101, 105, 115, 178, 177, 131, 150, 204, 187, 258, 299, 114, 109, 122, 77, 73, 122, 288, 333, 109, 135, 122, 146, 134, 157, 178, 168, 135, 79, 93, 65, 113, 127, 110, 133, 163, 86, 120, 157, 173, 67, 63, 85, 102, 95, 83, 89, 103, 71, 96, 173, 111, 122, 95, 117, 195, 100, 72, 75, 136, 87, 93, 185, 203, 148, 150, 150, 144, 143, 153, 101, 135, 109, 76, 77, 70, 61, 72, 106, 113, 114, 58, 50, 51, 76, 171, 177, 151, 143, 124, 118, 138, 138, 121, 157, 206, 183, 160, 202, 214, 195, 185, 52, 64, 61, 147, 126, 126, 105, 112, 109, 169, 264, 115, 93, 90, 75, 81, 73, 116, 119, 92, 90, 137, 72, 102, 90, 92, 66, 79, 95, 173, 148, 105, 99, 121, 79, 130, 135, 125, 120, 126, 131, 129, 118, 70, 71, 71, 80, 110, 112, 136, 302, 181, 181, 99, 78, 108, 132, 157, 153, 128, 139, 76, 100, 83, 96, 95, 62, 159, 113, 70, 83, 179, 148, 133, 119, 131, 114, 113, 124, 148, 138, 128, 148, 183, 64, 124, 128, 112, 108, 136, 161, 214, 96, 14

In [7]:
print(zip(widths_list,heights_list))

In [8]:
X = [[key,value] for key, value in zip(widths_list,heights_list)]


In [9]:
print(X)

[[104, 116], [108, 132], [82, 106], [125, 151], [89, 118], [117, 134], [75, 64], [121, 89], [111, 127], [79, 112], [55, 73], [61, 110], [53, 124], [100, 79], [89, 86], [77, 86], [51, 101], [48, 94], [61, 101], [69, 105], [68, 115], [147, 178], [201, 177], [121, 131], [117, 150], [115, 204], [118, 187], [99, 258], [130, 299], [74, 114], [85, 109], [183, 122], [93, 77], [96, 73], [186, 122], [146, 288], [198, 333], [67, 109], [61, 135], [84, 122], [114, 146], [60, 134], [89, 157], [108, 178], [79, 168], [54, 135], [62, 79], [61, 93], [45, 65], [61, 113], [63, 127], [57, 110], [65, 133], [83, 163], [121, 86], [113, 120], [171, 157], [182, 173], [80, 67], [49, 63], [71, 85], [79, 102], [81, 95], [59, 83], [61, 89], [82, 103], [59, 71], [50, 96], [116, 173], [66, 111], [82, 122], [76, 95], [91, 117], [130, 195], [97, 100], [46, 72], [44, 75], [75, 136], [42, 87], [53, 93], [78, 185], [84, 203], [72, 148], [71, 150], [83, 150], [67, 144], [69, 143], [110, 153], [75, 101], [61, 135], [126, 10

In [10]:
from sklearn.cluster import KMeans
import numpy as np
X = np.array(X)
kmeans = KMeans(n_clusters=4, random_state=0).fit(X)

In [11]:
kmeans.labels_

array([2, 2, 2, 1, 2, 1, 0, 2, 2, 2, 0, 2, 2, 0, 0, 0, 0, 0, 0, 2, 2, 1,
       1, 1, 1, 1, 1, 3, 3, 2, 2, 1, 0, 0, 1, 3, 3, 2, 2, 2, 1, 2, 1, 1,
       1, 2, 0, 0, 0, 2, 2, 2, 2, 1, 2, 2, 1, 1, 0, 0, 0, 2, 0, 0, 0, 2,
       0, 0, 1, 2, 2, 0, 2, 1, 2, 0, 0, 2, 0, 0, 1, 1, 2, 2, 2, 2, 2, 1,
       2, 2, 2, 0, 0, 0, 0, 0, 2, 2, 2, 0, 0, 0, 0, 1, 1, 1, 2, 2, 2, 2,
       2, 2, 1, 1, 1, 2, 1, 1, 1, 1, 0, 0, 0, 2, 2, 2, 2, 2, 0, 1, 3, 2,
       0, 0, 0, 0, 0, 2, 2, 0, 0, 2, 0, 0, 0, 2, 0, 0, 2, 1, 1, 0, 0, 2,
       0, 2, 2, 2, 2, 2, 2, 2, 2, 0, 0, 0, 0, 2, 2, 2, 3, 1, 1, 0, 0, 2,
       2, 1, 1, 2, 2, 0, 0, 0, 0, 0, 0, 2, 2, 0, 0, 1, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 1, 1, 0, 2, 2, 2, 0, 2, 1, 1, 0, 2, 0, 2, 3, 1, 0, 2,
       0, 0, 0, 0, 2, 0, 2, 0, 0, 3, 2, 0, 2, 1, 1, 1, 2, 2, 0, 2, 2, 1,
       1, 3, 1, 2, 2, 2, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 1, 0, 2, 0, 2, 0,
       0, 2, 0, 0, 0, 0, 0, 2, 0, 0, 1, 2, 2, 0, 1, 1, 0, 2, 2, 0, 2, 1,
       2, 3, 1, 0, 0, 1, 2, 1, 2, 3, 3, 1, 2, 2, 0,

In [12]:
kmeans.cluster_centers_

array([[ 54.568     ,  83.512     ],
       [106.796875  , 174.09375   ],
       [ 74.10869565, 124.11594203],
       [124.83333333, 263.83333333]])

### The  4 bounding box dimensions ([width, height]) are shown below:

- [ 54.568     ,  83.512     ]
- [106.796875  , 174.09375   ]
- [ 74.10869565, 124.11594203]
- [124.83333333, 263.83333333]

*Disclaimer: The contents of this notebook are used for educational purposes i.e. for learning and research.*